src/utils/parseError.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  errorCodes as MetaMaskErrorCodes,
+  getMessageFromCode,
+} from 'eth-rpc-errors'
+import {
+  LifiError,
+  LifiErrorCodes,
+  ProviderError,
+  RPCError,
+  ServerError,
+  UnknownError,
+  ValidationError,
+} from './errors'
+import { getChainById, Process, Step } from '@lifinance/types'
+import { formatTokenAmountOnly } from './utils'
+
+/**
+ * Available MetaMask error codes:
+ *
+ * export const errorCodes: ErrorCodes = {
+     rpc: {
+      invalidInput: -32000,
+      resourceNotFound: -32001,
+      resourceUnavailable: -32002,
+      transactionRejected: -32003,
+      methodNotSupported: -32004,
+      limitExceeded: -32005,
+      parse: -32700,
+      invalidRequest: -32600,
+      methodNotFound: -32601,
+      invalidParams: -32602,
+      internal: -32603,
+    },
+    provider: {
+      userRejectedRequest: 4001,
+      unauthorized: 4100,
+      unsupportedMethod: 4200,
+      disconnected: 4900,
+      chainDisconnected: 4901,
+    },
+  };
+ *
+ * For more information about error codes supported by metamask check
+ * https://github.com/MetaMask/eth-rpc-errors
+ * https://eips.ethereum.org/EIPS/eip-1474#error-codes
+ * https://eips.ethereum.org/EIPS/eip-1193#provider-errors
+ */
+
+const getTransactionNotSentMessage = (
+  step?: Step,
+  process?: Process
+): string => {
+  let transactionNotSend =
+    'Transaction was not sent, your funds are still in your wallet'
+
+  // add information about funds if available
+  transactionNotSend += step
+    ? ` (${formatTokenAmountOnly(
+        step.action.fromToken,
+        step.action.fromAmount
+      )} ${step.action.fromToken.symbol} on ${
+        getChainById(step.action.fromChainId).name
+      })`
+    : ''
+
+  transactionNotSend +=
+    ", please retry.<br/>If it still doesn't work, it is safe to delete this transfer and start a new one."
```

Comment:
I don't think this is a good idea. We should stay away from html in the sdk as it is also can be used on the server or any cmd line script.

<br> should be \n
The link should be rendered in the frontend

Reply:
IMO the text  for the process message is complete and holds enough info up to this point.
("[...] to delete this transfer and start a new one."). 

The sentence "You can check the failed transaction _here_" should be a frontend rendered thing.

Reply:
We need the html rendering here since we want to include the link to the transaction explorer which can not be done w/o HTML. 
It is not an issue if this is used on a server since this is just additional context information. The developer doesn't need to parse it, it is merely a comfort feature. 

I don't have a strong opinion on this but max suggested to move it to the sdk since it has more information about what exactly went wrong and where to look for solutions.